Formula/libxml2.rb

```diff
@@ -41,7 +41,7 @@ class Libxml2 < Formula
 
   keg_only :provided_by_osx
 
-  depends_on :python => :optional
+  depends_on :python if MacOS.version <= :snow_leopard
```

Comment:
It is unrelated to this PR; the built products of this formula will not be bound to any Python at all and may be imported by any Python 2.7. The situation for formulas like vim which embed a Python interpreter is very different than the situation for formulas that merely build bindings which may be imported from any compatible Python.

We are speaking about different homebrew-evolution proposals and I'm not sure which one you mean; the proposal for how to manage the many-pythons situation became unproductive and I abandoned it. The virtualenv proposal, which I consider very successful, was the only one of my proposals that was adopted.